backend/app/api/endpoints/users.py

```
@@ -292,6 +293,13 @@ async def create_new_user(user: UserCreate):
     try:
         created_user = await create_user(user_data)
         return created_user
+    except DuplicateKeyError:
+        # Concurrent insert slipped past the pre-checks and hit the unique index
+        # (issue #178) — surface the intended conflict, not a 500.
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail="User with this auth ID or email already exists",
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)